pkg/operator/ceph/object/user/controller.go

```diff
@@ -314,11 +318,15 @@ func (r *ReconcileObjectStoreUser) initializeObjectStoreContext(u *cephv1.CephOb
 	// Otherwise GetAdminOPSUserCredentials() will fail detecting the network provider when running RunAdminCommandNoMultisite()
 	objContext.CephClusterSpec = *r.cephClusterSpec
 
-	opsContext, err := object.NewMultisiteAdminOpsContext(objContext, &store.Spec)
-	if err != nil {
-		return errors.Wrap(err, "failed to initialized rgw admin ops client api")
+	// This is needed for the unit tests so the mock client does not get overridden
+	// This should not impact normal operations in production
+	if !IsRunningInUnitTest {
+		opsContext, err := object.NewMultisiteAdminOpsContext(objContext, &store.Spec)
+		if err != nil {
+			return errors.Wrap(err, "failed to initialized rgw admin ops client api")
+		}
+		r.objContext = opsContext
```

Comment:
A pattern we've used in the past that seems to me like it makes for slightly better-looking code is to be able to replace the `object.NewMultisiteAdminOpsContext()` for unit tests like I outline below. I think this implementation works, and I think we can merge it, but let me know if you like the below suggestion better.

```go
var newMultisiteAdminOpsCtxFunc = object.NewMultisiteAdminOpsContext

func main() {
    // ...
    opsCtx, err := newMultisiteAdminOpsCtxFunc(objContext, &store.Spec)
    // ...
}

---

func TestMain(t *testing.T) {
    newMultisiteAdminOpsCtxFunc = func(/* args */) /* returns */ {
        return /* a mock ctx */
    }
}
```

Reply:
I like this option better, much more elegant. Thanks!